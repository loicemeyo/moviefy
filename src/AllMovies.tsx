import React from 'react';
import axios from 'axios';
import styled from 'styled-components';
import MovieDetail from './MovieDetail';
import NoMovie from './NoMovie';

interface Rating {
    Source: string;
    Value: string;
}

export interface Movie {
    Title: string;
    Rated: string;
    Year: string;
    Released: string;
    Runtime: string;
    Genre: string;
    Director: string;
    Writer: string;
    Actors: string;
    Plot: string;
    Language: string;
    Country: string;
    Awards: string;
    Poster: string;
    Ratings: Rating[];
    Metascore: string;
    imdbRating: string;
    imdbVotes: string;
    imdbID: string;
    Type: string;
    DVD: string;
    BoxOffice: string;
    Production: string;
    Website: string;
    Response: string;
}

interface State {
    movies: Movie | null;
    title: string;
    noMovie: boolean;
}

class AllMovies extends React.Component<{}, State>{

    state:State = {
        movies: null,
        title: '',
        noMovie: false,
    };

    getMovies = async() => {
        const response = await axios.get(`http://www.omdbapi.com/?t=${this.state.title}=&apikey=40f8d155`);
        console.log(response);
        console.log(response.data.Error, 'error');
        if(!response.data.Error){
            this.setState({
                movies: response.data,
                noMovie: false
            })
        } else {
            this.setState({
                noMovie: true,
                movies: null
            })
        }
    }

    handleOnChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
        this.setState({
            title: event.target.value
        })
    }

    render(){
        const { movies, noMovie } = this.state; 
        return (
         <React.Fragment>
            <Header>
                <HeaderI>MOVIEFY</HeaderI>
                <HeaderII>...find your favorite movie and tv series</HeaderII>
                </Header>
            <Container>
            <Area>
            <Title><b>Movie Title:</b></Title>
            <Input type="text" onChange={this.handleOnChange}></Input>
            </Area>
            <Search onClick={this.getMovies}>Search</Search>
            </Container>
            {movies && <MovieDetail movieData={movies}></MovieDetail>}
            {!movies && noMovie && <NoMovie/>}
        </React.Fragment>
       )
       
    }

}

    const Header = styled.h1`
    text-align: center;
    font-family: 'Indie Flower', cursive;
    color: #D0CCCC;
    margin: 0 8px 2px 8px;
    background:
    /* top, transparent black, faked with gradient */ 
    linear-gradient(
        rgba(0, 0, 0, 0.7), 
        rgba(0, 0, 0, 0.7)
        ),
    /* bottom, image */
        url(http://fc02.deviantart.net/fs71/i/2011/274/6/f/ocean__sky__stars__and_you_by_muddymelly-d4bg1ub.png);
        }
    `;
    const HeaderI = styled.h1`
        font-size: 48px;
        margin-top: 0px;
        margin-bottom: 0px;
    `;

    const HeaderII = styled.h2`
        font-size: 20px;
        margin-top: 0px;
    `;


    export const Poster = styled.section`
    padding: 2px 2px;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    :hover {
        box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
      }
    `;
    
    const Container = styled.section`
    margin: 8px 24px;
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    `;

    const Area = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    `;


    const Title = styled.div`
    font-size: 30px
    font-family: 'Indie Flower', cursive;
    margin-right: 8px;
    `;

    const Input = styled.input`
    width: 50%;
    height: 25%
    padding: 8px 20px;
    margin-top: 0px;
    box-sizing: border-box;
    border-style: solid;
    `;

    const Search = styled.button`
    background-color: #555555;
    border-radius: 8px;
    margin-top: 5px;
    margin-left: 75px;
    border: none;
    color: white;
    padding: 8px 24px;
    text-align: center;
    align-items: center
    text-decoration: none;
    display: inline-block;
    font-size: 14px
    `;


export default AllMovies;