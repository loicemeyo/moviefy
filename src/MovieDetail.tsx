import React from 'react';
import styled from 'styled-components';
import { Poster,Movie } from './AllMovies';
import star from './star.svg'

const MovieDetail = (props: {movieData: Movie}) => {
    const { movieData } = props;
    return(
     <Detail>
         <Poster>
             <img src={movieData.Poster} />
         </Poster>
         <Info>
             <Title>
                 <b>{`${movieData.Title}`}</b> {`(${movieData.Year})`}  <img src={star} height={40} width={40} alt="logo" /> {`${movieData.imdbRating}`}
             </Title>
             <Row>
                 {`${movieData.Runtime} | ${movieData.Genre} | ${movieData.Released}`}
             </Row>
             <Plot>
                 {`${movieData.Plot}`}
             </Plot>
             <LastRow><b>Director:</b> {`${movieData.Director}`}</LastRow>
             <LastRow><b>Writer(s):</b> {`${movieData.Writer}`}</LastRow>
             <LastRow><b>Actors:</b> {`${movieData.Actors}`}</LastRow>
        </Info>
     </Detail>
    )
}

export default MovieDetail;

const Detail = styled.div`
    display: flex;
    align-items: center;
    justify-content: flex-start;

    @media (max-width: 768px) {
        flex-direction: column;
    }

`;

const Info = styled.div`
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    align-items: start;
    padding: 5px;

    @media (max-width: 768px) {
        padding: 30px;
    }
`;

const Title = styled.div`
    font-size: 32px;
    display: flex;
    align-items: center;
    text-align: start;
`;
const Row = styled.div`
    font-size: 12px;
    color: gray;
    text-align: start;
`;
const LastRow = styled.div`
    font-size: 14px;
    text-align: start;
`;

const Plot = styled.p`
    text-align: start;
`;

