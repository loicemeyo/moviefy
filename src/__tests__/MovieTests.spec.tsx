import React from 'react';
import { shallow, mount }from 'enzyme';
import AllMovies from '../AllMovies';
import MovieDetail from '../MovieDetail';
describe('AllMovies', () => {
  let wrapper: any;
  let mockOnchange: any;
  const movie = {
    Title: "Jumbo",
    Year: "2000",
    Rated: "string",
    Released: "string",
    Runtime: "string",
    Genre: "string",
    Director: "string",
    Writer: "string",
    Actors: "string",
    Plot: "string",
    Language: "string",
    Country: "string",
    Awards: "string",
    Poster: "string",
    Ratings: [],
    Metascore: "string",
    imdbRating: "string",
    imdbVotes: "string",
    imdbID: "string",
    Type: "string",
    DVD: "string",
    BoxOffice: "string",
    Production: "string",
    Website: "string",
    Response: "string"
  }

  beforeEach(() => {
    mockOnchange = jest.fn();
    wrapper = shallow(<AllMovies onChange={mockOnchange} />);
  });
  it("The component 'AllMovies' should render correctly", () => {
    const component = shallow(<AllMovies />);
  
    expect(component).toMatchSnapshot();
  });
  it("The component 'MovieDetail' should render correctly", () => {
    const component = shallow(<MovieDetail movieData={movie} />);
  
    expect(component).toMatchSnapshot();
  });
  it("should call handleOnChange", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleOnChange");
      
      wrapper.instance().forceUpdate();
      const mockEvent = {
        target: {
          name: "movie title",
          value: "jumbo"
        }
      };
      const expected = true;
      wrapper.instance().handleOnChange(mockEvent);
      expect(spy).toHaveBeenCalled();
    });
  it("check counter increment function is callled", () => {
      const wrapper:any = mount(<AllMovies />);
      const spy = jest.spyOn(wrapper.instance(), "getMovies");
      wrapper.setState({})
      wrapper.find("button").simulate("click");
      expect(spy).toHaveBeenCalled();
    });
  });