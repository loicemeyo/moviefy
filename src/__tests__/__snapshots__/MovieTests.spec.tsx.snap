// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllMovies The component 'AllMovies' should render correctly 1`] = `
<Fragment>
  <styled.h1>
    <styled.h1>
      MOVIEFY
    </styled.h1>
    <styled.h2>
      ...find your favorite movie and tv series
    </styled.h2>
  </styled.h1>
  <styled.section>
    <styled.div>
      <styled.div>
        <b>
          Movie Title:
        </b>
      </styled.div>
      <styled.input
        onChange={[Function]}
        type="text"
      />
    </styled.div>
    <styled.button
      onClick={[Function]}
    >
      Search
    </styled.button>
  </styled.section>
</Fragment>
`;

exports[`AllMovies The component 'MovieDetail' should render correctly 1`] = `
<styled.div>
  <styled.section>
    <img
      src="string"
    />
  </styled.section>
  <styled.div>
    <styled.div>
      <b>
        Jumbo
      </b>
       
      (2000)
        
      <img
        alt="logo"
        height={40}
        src="star.svg"
        width={40}
      />
       
      string
    </styled.div>
    <styled.div>
      string | string | string
    </styled.div>
    <styled.p>
      string
    </styled.p>
    <styled.div>
      <b>
        Director:
      </b>
       
      string
    </styled.div>
    <styled.div>
      <b>
        Writer(s):
      </b>
       
      string
    </styled.div>
    <styled.div>
      <b>
        Actors:
      </b>
       
      string
    </styled.div>
  </styled.div>
</styled.div>
`;
